src/ios/LocalyticsPlugin.m

```diff
@@ -423,191 +427,286 @@ - (void)updateInboxCampaignCache {
+    [self logInput:@"autoIntegrate" withCommand:command];
     NSString *appKey = nil;
+    NSDictionary *localyticsOptions = nil;
+
     if ([command argumentAtIndex:0]) {
         appKey = [command argumentAtIndex:0];
     } else {
         appKey = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"LocalyticsAppKey"];
     }
 
+    if ([command argumentAtIndex:1]) {
+      localyticsOptions = [command argumentAtIndex:1];
+    } else {
+      NSLog(@"Localytics Cordova wrapper can't integrate. No App Key found.");
+    }
+
     if (appKey) {
-        localyticsIsAutoIntegrate = YES;
-        [Localytics autoIntegrate:appKey launchOptions: launchOptions];
-        launchOptions = nil; // Clear launchOptions on integrate
+      localyticsIsAutoIntegrate = YES;
+      [Localytics autoIntegrate:appKey withLocalyticsOptions:localyticsOptions launchOptions: launchOptions];
+      launchOptions = nil; // Clear launchOptions on integrate
+    } else {
+      NSLog(@"Localytics Cordova wrapper can't integrate. No App Key found.");
     }
 }
 
 - (void)openSession:(CDVInvokedUrlCommand *)command {
+    [self logInput:@"openSession" withCommand:command];
     [Localytics openSession];
 }
 
 - (void)closeSession:(CDVInvokedUrlCommand *)command {
+    [self logInput:@"closeSession" withCommand:command];
     [Localytics closeSession];
 }
 
 - (void)upload:(CDVInvokedUrlCommand *)command {
+    [self logInput:@"upload" withCommand:command];
     [Localytics upload];
 }
 
+- (void)pauseDataUploading:(CDVInvokedUrlCommand *)command {
+    [self logInput:@"pauseDataUploading" withCommand:command];
+    NSNumber *pause = [command argumentAtIndex:0];
+    if ([pause isKindOfClass:[NSNumber class]]) {
+      [Localytics pauseDataUploading:[pause boolValue]];
+    } else {
+      NSLog(@"Localytics Cordova wrapper received bad input in call to pauseDataUploading.");
+    }
+}
+
 
 #pragma mark Analytics
 
 - (void)setOptedOut:(CDVInvokedUrlCommand *)command {
+    [self logInput:@"setOptedOut" withCommand:command];
     NSNumber *enabled = [command argumentAtIndex:0];
     if ([enabled isKindOfClass:[NSNumber class]]) {
-        [Localytics setOptedOut:[enabled boolValue]];
+      [Localytics setOptedOut:[enabled boolValue]];
+    } else {
+      NSLog(@"Localytics Cordova wrapper received bad input in call to setOptedOut.");
     }
 }
 
 - (void)isOptedOut:(CDVInvokedUrlCommand *)command {
+    [self logInput:@"isOptedOut" withCommand:command];
     [self.commandDelegate runInBackground:^{
         BOOL value = [Localytics isOptedOut];
         CDVPluginResult *pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsBool:value];
         [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
     }];
 }
 
+- (void)setPrivacyOptedOut:(CDVInvokedUrlCommand *)command {
+    [self logInput:@"setPrivacyOptedOut" withCommand:command];
+    NSNumber *enabled = [command argumentAtIndex:0];
```

Comment:
Why is this an NSNumber?

Reply:
Those arguments are converted to objects, so we decode as an `NSNumber` and then make sense of it as a boolean